# nf-core

:::: {tab-set}

::: {tab-item} 2.10

[![](badges/release-2.10-blue.svg)](https://cloud.sdu.dk/app/applications?tag=nf-core&itemsPerPage=100&page=0)
![type](badges/type-batch-yellow.svg)
![access](badges/access-open-green.svg)
* **Operating System:** ![](./badges/Ubuntu-22.04-lightseagreen.svg)
* **Terminal:** ![](./badges/tini-0.19.0-lightseagreen.svg) ![](./badges/tmux-3.2a-lightseagreen.svg)
* **Shell:** ![](./badges/bash-5.1.16-lightseagreen.svg) ![](./badges/fish-3.3.1-lightseagreen.svg) ![](./badges/zsh-5.8.1-lightseagreen.svg)
* **Editor:** ![](./badges/emacs-27.1-lightseagreen.svg) ![](./badges/nano-6.2-lightseagreen.svg) ![](./badges/vim-8.2-lightseagreen.svg)
* **Package Manager:** ![](./badges/apt-2.4.10-lightseagreen.svg) ![](./badges/conda-23.3.1-lightseagreen.svg) ![](./badges/dpkg-1.21.1-lightseagreen.svg) ![](./badges/npm-8.5.1-lightseagreen.svg) ![](./badges/pip-23.2.1-lightseagreen.svg)
* **Programming Language:** ![](./badges/GCC-11.4.0-lightseagreen.svg) ![](./badges/Python-3.10.12-lightseagreen.svg)
* **Utility:** ![](./badges/EasyBuild-4.8.0-lightseagreen.svg) ![](./badges/Lmod-8.7-lightseagreen.svg)
* **Extension:** ![](./badges/OpenMPI-4.1.2-lightseagreen.svg)

:::

::: {tab-item} 2.9

[![](badges/release-2.9-blue.svg)](https://cloud.sdu.dk/app/applications?tag=nf-core&itemsPerPage=100&page=0)
![type](badges/type-batch-yellow.svg)
![access](badges/access-open-green.svg)
* **Operating System:** ![](./badges/Ubuntu-22.04-lightseagreen.svg)
* **Terminal:** ![](./badges/tini-0.19.0-lightseagreen.svg) ![](./badges/tmux-3.2a-lightseagreen.svg)
* **Shell:** ![](./badges/bash-5.1.16-lightseagreen.svg) ![](./badges/fish-3.3.1-lightseagreen.svg) ![](./badges/zsh-5.8.1-lightseagreen.svg)
* **Editor:** ![](./badges/emacs-27.1-lightseagreen.svg) ![](./badges/nano-6.2-lightseagreen.svg) ![](./badges/vim-8.2-lightseagreen.svg)
* **Package Manager:** ![](./badges/apt-2.4.10-lightseagreen.svg) ![](./badges/conda-23.3.1-lightseagreen.svg) ![](./badges/dpkg-1.21.1-lightseagreen.svg) ![](./badges/npm-8.5.1-lightseagreen.svg) ![](./badges/pip-23.2.1-lightseagreen.svg)
* **Programming Language:** ![](./badges/GCC-11.4.0-lightseagreen.svg) ![](./badges/Python-3.10.12-lightseagreen.svg)
* **Utility:** ![](./badges/EasyBuild-4.8.0-lightseagreen.svg) ![](./badges/Lmod-8.7-lightseagreen.svg)
* **Extension:** ![](./badges/OpenMPI-4.1.2-lightseagreen.svg)

:::

::: {tab-item} 2.8

[![](badges/release-2.8-blue.svg)](https://cloud.sdu.dk/app/applications?tag=nf-core&itemsPerPage=100&page=0)
![type](badges/type-batch-yellow.svg)
![access](badges/access-open-green.svg)
* **Operating System:** ![](./badges/Ubuntu-22.10-lightseagreen.svg)
* **Shell:** ![](./badges/bash-5.2.2-lightseagreen.svg) ![](./badges/fish-3.5.1-lightseagreen.svg) ![](./badges/zsh-5.9-lightseagreen.svg)
* **Editor:** ![](./badges/emacs-27.1-lightseagreen.svg) ![](./badges/nano-6.4-lightseagreen.svg) ![](./badges/vim-9.0-lightseagreen.svg)
* **Package Manager:** ![](./badges/apt-2.5.3-lightseagreen.svg) ![](./badges/conda-23.1.0-lightseagreen.svg)
![](./badges/dpkg-1.21.9-lightseagreen.svg) ![](./badges/pip-23.0.1-lightseagreen.svg)
* **Programming Language:** ![](./badges/GCC-12.2.0-lightseagreen.svg) ![](./badges/OpenJDK-17.0.3-lightseagreen.svg) ![](./badges/Python-3.10.10-lightseagreen.svg)
:::

::: {tab-item} 2.7.2

[![](badges/release-2.7.2-blue.svg)](https://cloud.sdu.dk/app/applications?tag=nf-core&itemsPerPage=100&page=0)
![type](badges/type-batch-yellow.svg)
![access](badges/access-open-green.svg)
* **Operating System:** ![](./badges/Ubuntu-22.10-lightseagreen.svg)
* **Shell:** ![](./badges/bash-5.2.2-lightseagreen.svg) ![](./badges/fish-3.5.1-lightseagreen.svg) ![](./badges/zsh-5.9-lightseagreen.svg)
* **Editor:** ![](./badges/emacs-27.1-lightseagreen.svg) ![](./badges/nano-6.4-lightseagreen.svg) ![](./badges/vim-9.0-lightseagreen.svg)
* **Package Manager:** ![](./badges/apt-2.5.3-lightseagreen.svg) ![](./badges/conda-23.1.0-lightseagreen.svg) ![](./badges/dpkg-1.21.9-lightseagreen.svg) ![](./badges/pip-23.0.1-lightseagreen.svg)
* **Programming Language:** ![](./badges/GCC-12.2.0-lightseagreen.svg) ![](./badges/OpenJDK-17.0.3-lightseagreen.svg) ![](./badges/Python-3.10.10-lightseagreen.svg)
:::

::::

`nf-core` is a collection of analysis pipelines built using Nextflow.

For more information, check [here](https://nf-co.re/).

## Supported pipelines

| **Pipeline** | **Release** | **nf-core** | **Description** |
|---|---|---|---|
| ``airrflow`` | [![](badges/release-3.0-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-airrflow&version=3.0) [![](badges/release-3.1.0-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-airrflow&version=3.1.0) | ![](badges/release-2.7.2-blue.svg) ![](badges/release-2.8-blue.svg) | Define where the pipeline should find input data and save output data. |
| ``ampliseq`` | [![](badges/release-2.6.1-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-ampliseq&version=2.6.1) [![](badges/release-2.7.1-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-ampliseq&version=2.7.1) | ![](badges/release-2.9-blue.svg) ![](badges/release-2.10-blue.svg) | Bioinformatics analysis pipeline used for amplicon sequencing. |
| ``atacseq`` | [![](badges/release-2.0-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-atacseq&version=2.0) [![](badges/release-2.1.2-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-atacseq&version=2.1.2) | ![](badges/release-2.7.2-blue.svg) ![](badges/release-2.9-blue.svg) | Bioinformatics analysis pipeline used for ATAC-seq data. |
| ``bacass`` | [![](badges/release-2.0.0-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-bacass&version=2.0.0) [![](badges/release-2.1.0-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-bacass&version=2.1.0) | ![](badges/release-2.9-blue.svg) ![](badges/release-2.10-blue.svg) | Bioinformatics best-practice analysis pipeline for simple bacterial assembly and annotation. |
| ``bamtofastq`` | [![](badges/release-2.0.0-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-bamtofastq&version=2.0.0) [![](badges/release-2.1.0-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-bamtofastq&version=2.1.0) | ![](badges/release-2.9-blue.svg) ![](badges/release-2.9-blue.svg) | Workflow designed to convert one or multiple bam/cram files into fastq format. |
| ``chipseq`` | [![](badges/release-2.0.0-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-chipseq&version=2.0.0) | ![](badges/release-2.9-blue.svg) | Bioinformatics analysis pipeline used for Chromatin ImmunopreciPitation sequencing (ChIP-seq) data. |
| ``circdna`` | [![](badges/release-1.0.4-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-circdna&version=1.0.4) | ![](badges/release-2.9-blue.svg) | Bioinformatics best-practice analysis pipeline for the identification of extrachromosomal circular DNAs (ecDNAs) in eukaryotic cells. |
| ``crisprseq`` | [![](badges/release-1.0-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-crisprseq&version=1.0) [![](badges/release-2.0.0-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-crisprseq&version=2.0.0) [![](badges/release-2.1.0-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-crisprseq&version=2.1.0) | ![](badges/release-2.7.2-blue.svg) ![](badges/release-2.9-blue.svg) ![](badges/release-2.10-blue.svg) | Bioinformatics best-practice analysis pipeline for the analysis of CRISPR edited next generation sequencing (NGS) data, it allows the evaluation of the quality of gene editing experiments using targeted NGS data. |
| ``cutandrun`` | [![](badges/release-3.1-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-cutandrun&version=3.1) [![](badges/release-3.2.1-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-cutandrun&version=3.2.1) | ![](badges/release-2.9-blue.svg) ![](badges/release-2.10-blue.svg) | Best-practice bioinformatic analysis pipeline for CUT&RUN and CUT&Tag experimental protocols that were developed to study protein-DNA interactions and epigenomic profiling. |
| ``differentialabundance`` | [![](badges/release-1.1.1-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-differentialabundance&version=1.1.1) [![](badges/release-1.2.0-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-differentialabundance&version=1.2.0) [![](badges/release-1.3.1-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-differentialabundance&version=1.3.1) | ![](badges/release-2.7.2-blue.svg) ![](badges/release-2.7.2-blue.svg) ![](badges/release-2.10-blue.svg) | Bioinformatics pipeline that can be used to analyze data represented as matrices, comparing groups of observations to generate differential statistics and downstream analyses. The initial feature set is built around RNA-seq, but we anticipate rapid expansion to include other platforms.  |
| ``epitopeprediction`` | [![](badges/release-2.2.1-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-epitopeprediction&version=2.2.1) | ![](badges/release-2.9-blue.svg) | Bioinformatics best-practice analysis pipeline for epitope prediction and annotation. |
| ``fetchngs`` | [![](badges/release-1.10.1-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-fetchngs&version=1.10.1) [![](badges/release-1.11.0-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-fetchngs&version=1.11.0) | ![](badges/release-2.9-blue.svg) ![](badges/release-2.9-blue.svg) | Bioinformatics best-practice analysis pipeline to fetch metadata and raw FastQ files from both public and private databases. At present, the pipeline supports SRA / ENA / DDBJ / GEO / Synapse ids. |
| ``funcscan`` | [![](badges/release-1.1.4-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-funcscan&version=1.1.4) | ![](badges/release-2.10-blue.svg) | Bioinformatics pipeline for efficient and parallelised screening of long nucleotide sequences such as contigs for antimicrobial peptide genes, antimicrobial resistance genes, and biosynthetic gene clusters. |
| ``hgtseq`` | [![](badges/release-1.1.0-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-hgtseq&version=1.1.0) | ![](badges/release-2.8-blue.svg) | Bioinformatics best-practice analysis pipeline built to investigate horizontal gene transfer from NGS data. |
| ``hic`` | [![](badges/release-2.1.0-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-hic&version=2.1.0) | ![](badges/release-2.9-blue.svg) | Bioinformatics best-practice analysis pipeline for Analysis of Chromosome Conformation Capture data (Hi-C). |
| ``hicar`` | [![](badges/release-1.0.0-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-hicar&version=1.0.0) | ![](badges/release-2.9-blue.svg) | Bioinformatics best-practice analysis pipeline for HiC on Accessible Regulatory DNA (HiCAR) data, a robust and sensitive assay for simultaneous measurement of chromatin accessibility and cis-regulatory chromatin contacts. |
| ``hlatyping`` | [![](badges/release-2.0.0-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-hlatyping&version=2.0.0) | ![](badges/release-2.9-blue.svg) | Bioinformatics best-practice analysis pipeline for Precision HLA typing from next-generation sequencing data. |
| ``isoseq`` | [![](badges/release-1.1.5-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-isoseq&version=1.1.5) | ![](badges/release-2.9-blue.svg) | Bioinformatics best-practice analysis pipeline for Isoseq gene annotation with uLTRA and TAMA. |
| ``mag`` | [![](badges/release-2.3.2-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-mag&version=2.3.2) [![](badges/release-2.4.0-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-mag&version=2.4.0) [![](badges/release-2.5.1-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-mag&version=2.5.1)| ![](badges/release-2.9-blue.svg) ![](badges/release-2.9-blue.svg) ![](badges/release-2.10-blue.svg) | Bioinformatics best-practise analysis pipeline for assembly, binning and annotation of metagenomes. |
| ``methylseq`` | [![](badges/release-2.4.0-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-methylseq&version=2.4.0) [![](badges/release-2.5.0-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-methylseq&version=2.5.0) | ![](badges/release-2.9-blue.svg) ![](badges/release-2.9-blue.svg) | Bioinformatics analysis pipeline used for Methylation (Bisulfite) sequencing data. It pre-processes raw data from FastQ inputs, aligns the reads and performs extensive quality-control on the results. |
| ``mhcquant`` | [![](badges/release-2.4.1-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-mhcquant&version=2.4.1) [![](badges/release-2.5.0-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-mhcquant&version=2.5.0) | ![](badges/release-2.9-blue.svg) ![](badges/release-2.9-blue.svg) | Bioinformatics analysis pipeline used for quantitative processing of data dependent (DDA) peptidomics data. |
| ``nanostring`` | [![](badges/release-1.1.1-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-nanostring&version=1.1.1) [![](badges/release-1.2.0-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-nanostring&version=1.2.0) | ![](badges/release-2.9-blue.svg) ![](badges/release-2.9-blue.svg) | Bioinformatics pipeline that can be used to analyze NanoString data. The performed analysis steps include quality control and data normalization. |
| ``nascent`` | [![](badges/release-2.1.1-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-nascent&version=2.1.1) | ![](badges/release-2.8-blue.svg) | Bioinformatics best-practice analysis pipeline for nascent transcript (NT) and Transcriptional Start Site (TSS) assays. |
| ``pangenome`` | [![](badges/release-1.0.0-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-pangenome&version=1.0.0) | ![](badges/release-2.9-blue.svg) | Bioinformatics best-practice analysis pipeline for pangenome graph construction. The pipeline renders a collection of sequences into a pangenome graph. |
| ``phyloplace`` | [![](badges/release-1.0.0-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-phyloplace&version=1.0.0) | ![](badges/release-2.7.2-blue.svg) | Bioinformatics best-practice analysis pipeline that performs phylogenetic placement with EPA-NG. |
| ``proteinfold`` | [![](badges/release-1.0.0-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-proteinfold&version=1.0.0) | ![](badges/release-2.7.2-blue.svg) | Bioinformatics best-practice analysis pipeline for Protein 3D structure prediction pipeline. |
| ``quantms`` | [![](badges/release-1.1.2-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-quantms&version=1.1.2) [![](badges/release-1.2.0-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-quantms&version=1.2.0) | ![](badges/release-2.9-blue.svg) ![](badges/release-2.10-blue.svg) | Bioinformatics best-practice analysis pipeline for Quantitative Mass Spectrometry (MS). Currently, the workflow supports three major MS-based analytical methods: (i) Data dependant acquisition (DDA) label-free and Isobaric quantitation (e.g. TMT, iTRAQ); (ii) Data independent acquisition (DIA) label-free quantification. |
| ``rnafusion`` | [![](badges/release-2.3.4-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-rnaseq&version=2.3.4) [![](badges/release-2.4.0-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-rnaseq&version=2.4.0) | ![](badges/release-2.9-blue.svg) ![](badges/release-2.9-blue.svg) | Bioinformatics best-practice analysis pipeline for RNA sequencing analysis pipeline with curated list of tools for detecting and visualizing fusion genes. |
| ``rnaseq`` | [![](badges/release-3.11.2-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-rnaseq&version=3.11.2) [![](badges/release-3.12.0-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-rnaseq&version=3.12.0) [![](badges/release-3.13.1-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-rnaseq&version=3.13.1) | ![](badges/release-2.7.2-blue.svg) ![](badges/release-2.8-blue.svg) ![](badges/release-2.10-blue.svg) | Bioinformatics pipeline that can be used to analyze RNA sequencing data obtained from organisms with a reference genome and annotation. |
| ``sarek`` | [![](badges/release-3.2.3-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-sarek&version=3.2.3) [![](badges/release-3.3.2-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-sarek&version=3.3.2) [![](badges/release-3.4.0-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-sarek&version=3.4.0) | ![](badges/release-2.9-blue.svg) ![](badges/release-2.9-blue.svg) ![](badges/release-2.10-blue.svg) | Workflow designed to detect variants on whole genome or targeted sequencing data. Initially designed for Human, and Mouse, it can work on any species with a reference genome. Sarek can also handle tumour / normal pairs and could include additional relapses. |
| ``scrnaseq`` | [![](badges/release-2.3.2-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-scrnaseq&version=2.3.2) [![](badges/release-2.4.1-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-scrnaseq&version=2.4.1) | ![](badges/release-2.9-blue.svg) ![](badges/release-2.9-blue.svg) | Bioinformatics best-practice analysis pipeline for processing 10x Genomics single-cell RNA-seq data. |
| ``smrnaseq`` | [![](badges/release-2.2.4-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-smrnaseq&version=2.2.4) | ![](badges/release-2.10-blue.svg) | Bioinformatics best-practice analysis pipeline for Small RNA-Seq. |
| ``taxprofiler`` | [![](badges/release-1.0.1-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-taxprofiler&version=1.0.1) [![](badges/release-1.1.2-blue.svg)](https://cloud.sdu.dk/app/jobs/create?app=nf-core-taxprofiler&version=1.1.2) | ![](badges/release-2.9-blue.svg) ![](badges/release-2.10-blue.svg) | Analysis pipeline for taxonomic classification and profiling of shotgun metagenomic data. It allows for in-parallel taxonomic identification of reads or taxonomic abundance estimation with multiple classification and profiling tools against multiple databases, produces standardised output tables. |

## Batch mode

Nextflow has built-in support for Conda that allows the configuration of workflow dependencies using Conda recipes and environment files.

In batch mode all pipelines are run by default with the option *-profile conda*.
In addition, for each pipeline, the necessary Conda environments are *pre-installed* in the app container.
The Conda cache directory is set as ``$NXF_CONDA_CACHEDIR=/home/ucloud/.cache/nextflow``.

``` {note}
On UCloud is not possible to use *singularity* or *docker* profiles.
However most of the pipelines support Conda/Mamba.
```

## Interactive mode

The _Interactive mode_ parameter is used to start an interactive job session where the user can open a terminal interface from the job progress page and execute shell commands.

Pipelines are executed as follows:

```console
$ nextflow run nf-core/<pipeline> -r <release> <options>
```

## Pipeline testing

In interactive mode you can test any nf-core pipeline with the following commands:

```console
$ nextflow run nf-core/<pipeline_name> -r <release> -profile test,conda
```

or

```console
$ nextflow run nf-core/<pipeline_name> -r <release> -profile test,mamba
```
